src/adapters/communities-db.ts

```diff
@@ -1092,6 +1092,36 @@ export function createCommunitiesDBComponent(
         DELETE FROM community_post_likes
         WHERE post_id IN (SELECT id FROM community_posts WHERE community_id = ${communityId}) AND user_address = ${normalizedAddress}
       `)
+    },
+
+    async updateCommunityRankingScore(communityId: string, score: number): Promise<void> {
+      await pg.query(SQL`
+        UPDATE communities
+        SET ranking_score = ${score}, last_score_calculated_at = now()
+        WHERE id = ${communityId}
+      `)
+    },
+
+    async setEditorChoice(communityId: string, isEditorChoice: boolean): Promise<void> {
+      await pg.query(SQL`
+        UPDATE communities
+        SET editors_choice = ${isEditorChoice}
+        WHERE id = ${communityId}
+      `)
+    },
```

Comment:
NIT:

We could also merge these two updates with conditional updating if you wish!